### DISCUSSION TITLE
Form Values not Rendering on Update

### DISCUSSION THREAD

**Discussion by HunterJuneau** [nss-evening-cohort-14/evening-client-side#341] [Capstones!]
Hunter Juneau

I couldn't find an existing ticket for this.

My update values are not being shown in my form. According to React dev tools, the data is being passed, but my state is showing undefined for all the values.

A snippet of my AchievementForm
```
export default function AchievementForm({ userId, data }) {
  const [achievement, setAchievement] = useState({
    name: data ? data.name : '',
    img: data ? data.img : '',
    description: data ? data.description : '',
    achieved: data ? data.achieved : false,
    gameKey: data ? data.gameKey : '',
    uid: userId,
    key: data ? data.key : null,
  });
```

[Link to Code](https://github.com/HunterJuneau/achievement-tracker/blob/update-achievements/src/components/forms/AchievementForm.js)

**Comment by Gabrielle-Tobermann** [comment]
Did you try just doing ```name: data.name || ''``` ?

**Reply by Gabrielle-Tobermann** [reply]
You also might wanna try passing name, img, description etc as props instead pf passing the data object

**Reply by HunterJuneau** [reply]
@Gabrielle-Tobermann  I have a strange result.

As I was editing the code to your suggestions, the DOM was re-rendering and appeared to be working. Fields were filled out and everything. But then, I went back to test it and it didn't work. I clicked the edit button on my achievements and the fields were empty. So I'm not sure what that means.

**Reply by HunterJuneau** [reply]
I think the issue has something to do with the form not re-rendering when I get the data response from Firebase.
